bot/exts/valentines/valentine_zodiac.py

```diff
@@ -19,36 +22,116 @@ class ValentineZodiac(commands.Cog):
+        explanation_file = Path("bot/resources/valentines/zodiac_explanation.json")
+        compatibility_file = Path("bot/resources/valentines/zodiac_compatibility.json")
+        with explanation_file.open(encoding="utf8") as json_data:
+            zodiac_fact = load(json_data)
+            for _, zodiac_data in zodiac_fact.items():
+                zodiac_data['start_at'] = datetime.fromisoformat(zodiac_data['start_at'])
+                zodiac_data['end_at'] = datetime.fromisoformat(zodiac_data['end_at'])
+        with compatibility_file.open(encoding="utf8") as json_data:
             zodiacs = load(json_data)
-            return zodiacs
+        return zodiacs, zodiac_fact
 
-    @commands.command(name="partnerzodiac")
-    async def counter_zodiac(self, ctx: commands.Context, zodiac_sign: str) -> None:
+    def zodiac_sign_verify(self, zodiac: str) -> discord.Embed:
+        """Gives informative zodiac embed."""
+        zodiac = zodiac.capitalize()
+        embed = discord.Embed()
+        embed.color = Colours.pink
+        if zodiac.capitalize() in self.zodiac_fact:
+            log.info("Making zodiac embed")
+            embed.title = f"__{zodiac}__"
+            embed.description = self.zodiac_fact[zodiac]["About"]
+            embed.add_field(name='__Full form__', value=self.zodiac_fact[zodiac]["full_form"], inline=False)
+            embed.add_field(name='__Motto__', value=self.zodiac_fact[zodiac]["Motto"], inline=False)
+            embed.add_field(name='__Strengths__', value=self.zodiac_fact[zodiac]["Strengths"], inline=False)
+            embed.add_field(name='__Weaknesses__', value=self.zodiac_fact[zodiac]["Weaknesses"], inline=False)
+            embed.set_thumbnail(url=self.zodiac_fact[zodiac]["url"])
+        else:
+            err_comp = [f"`{i}` {zod_name}" for i, zod_name in enumerate(self.zodiac_fact.keys(), start=1)]
+            error = ("\n").join(err_comp)
+            error_msg = f"`{zodiac}` is not a valid zodiac sign, here is the list of valid zodiac signs."
+            embed.description = f"{error_msg}\n{error}"
+            log.info("Wrong Zodiac name provided")
+        log.info("Zodiac embed ready")
+        return embed
+
+    def zodiac_date_verifier(self, query_datetime: datetime) -> str:
+        """Returns zodiac sign by checking month and date."""
+        for zodiac_name, zodiac_data in self.zodiac_fact.items():
+            if zodiac_data["start_at"].date() <= query_datetime.date() <= zodiac_data["end_at"].date():
+                zodiac = zodiac_name
+        log.info("Zodiac name sent")
+        return zodiac
+
+    @commands.group(name='zodiac', invoke_without_command=True)
+    async def zodiac(self, ctx: commands.Context, zodiac_sign: str) -> None:
+        """Provides information about zodiac sign by taking zodiac sign name as input."""
+        if zodiac_sign.startswith("`"):
+            await ctx.send("Please don't include `")
+            return
```

Comment:
What is the need of this? I don't think users will generally use backticks.

Reply:
Actually if user search for ` the embed looks weired that y I have included this